app/helpers/database_service.py

```diff
@@ -0,0 +1,156 @@
+
+    # Create or check user exists database
+    def create_db(self, db_name=None, user=None, password=None):
+        try:
+            connection = self.create_connection()
+            if not connection:
+                return False
+            cursor = connection.cursor()
+            cursor.execute(f"CREATE DATABASE {db_name}")
+            if self.create_user(user=user, password=password):
+                cursor.execute(
+                    f"GRANT ALL PRIVILEGES ON {db_name}.* To '{user}'")
+            return True
+        except Error as e:
+            print(e)
+            return False
+        finally:
+            if not connection:
+                return False
+            if (connection.is_connected()):
+                cursor.close()
+                connection.close()
+
+    # create database user
+    def create_user(self, user=None, password=None):
+        try:
+            connection = self.create_connection()
+            if not connection:
+                return False
+            cursor = connection.cursor()
+            cursor.execute(
+                f"CREATE USER '{user}' IDENTIFIED BY '{password}' ")
+            return True
+        except Error as e:
+            if e.errno == '1396':
+                return True
+            return False
+        finally:
+            if not connection:
+                return False
+            if (connection.is_connected()):
+                cursor.close()
+                connection.close()
+
+    # delete database user
+    def delete_user(self, user=None):
+        try:
+            connection = self.create_connection()
+            if not connection:
+                return False
+            cursor = connection.cursor()
+            cursor.execute(f"DROP USER '{user}' ")
+            return True
+        except Error:
+            return False
+        finally:
+            if not connection:
+                return False
+            if (connection.is_connected()):
+                cursor.close()
+                connection.close()
+
+    # delete database
+
+    def delete_db(self, db_name):
+        try:
+            connection = self.create_connection()
+            if not connection:
+                return False
+            cursor = connection.cursor()
+            cursor.execute(f"DROP DATABASE {db_name}")
+            return True
+        except Error:
+            return False
+        finally:
+            if not connection:
+                return False
+            if (connection.is_connected()):
+                cursor.close()
+                connection.close()
+
+    # Show all databases
+
+    def get_all_databases(self):
+        try:
+            connection = self.create_connection()
+            if not connection:
+                return False
+            cursor = connection.cursor()
+            cursor.execute("SHOW DATABASES")
+            database_list = []
+            for db in cursor:
+                database_list.append(db[0].decode())
```

Comment:
When i tested the get_all_databases method, it threw an error about the decode method not being needed. when i removed it that's when it worked! 

![Screenshot from 2021-02-19 16-08-41](https://user-images.githubusercontent.com/36065782/108510104-2a652300-72cf-11eb-9eae-724a904d2ca0.png)

Reply:
It was just a difference in environments for testing.